### PR TITLE
Fix Loop Mt. Moon 1F - Route 3

### PR DESCRIPTION
**Pull request by ghost**
Fixes issue #4.

Hope it helps ;)

### CODE REVIEW

**Comment by g0ldPRO** [comment]
If I recall correctly, I made that quest before implementing a generic `Quest:needPokecenter()` that would do all the necessary checks.

**Comment by ghost** [comment]
Yes, when the training is done in Mt. Moon 1F and it goes back to pc, the bot moves back to Route 3, I guess the boolean `self.healPokemonOnceTrainingIsOver` returns "true" so it sends it back to Route 3, and `function MoonFossilQuest:Route3()` sends it back to Mt. Moon 1F creating a loop so I fixed it in the MoonFossilQuest code.

But I get what you mean, Iguess the boolean `self.healPokemonOnceTrainingIsOver` in `Quest:needPokecenter()` should return "false" after using PC when training is done; so the fix should be done there.

File: Quests/Kanto/MoonFossilQuest.lua [hunk 1]
```diff
@@ -54,7 +54,8 @@ function MoonFossilQuest:MtMoon1F()
 		return talkToNpcOnCell(19, 16) -- moon stone
 	elseif not self:isTrainingOver() then
```
Comment (ghost) [inline Quests/Kanto/MoonFossilQuest.lua:55]: Due to this line, the `self.healPokemonOnceTrainingIsOver` will always return "true" and the bot loops.
Not sure how to fix it tho.

**Comment by g0ldPRO** [comment]
Fixed with 78781e2

**Pull request closed without merging**